src/main/java/trackitnus/logic/parser/ParserUtil.java

```diff
@@ -143,6 +144,25 @@ public static Code parseCode(String code) throws ParseException {
         return new Code(trimmedCode);
     }
 
+    /**
+     * Parses an {@code Optional<String> code} into an {@code Code}.
+     *
+     * @throws ParseException if the given {@code code} is invalid.
+     */
+    public static Code parseOptionalCode(Optional<String> code) throws ParseException {
+        if (code.isEmpty()) {
+            return null;
+        }
+        String trimmedCode = code.get().trim();
+        if (!Task.isValidString(trimmedCode)) {
+            return null;
+        }
+        if (!Code.isValidCode(trimmedCode)) {
+            throw new ParseException(Code.MESSAGE_CONSTRAINTS);
+        }
```

Comment:
why do you need to check Task.isValidString here? I thought only the latter check is necessary?

Reply:
This is to support the feature that when the user types `c/` it will reset the code to null, in other words they can remove the module code. If only have the latter check `c/` will throw error